supabase/models/messaging/startNewConversation.ts

```diff
@@ -62,13 +66,31 @@ export default async function startNewConversation(
         'This is the start of your conversation.'
       );
 
-      return conversationId;
+      const { data: newUserConversation, error: newUserConversationError } =
+        await supabase
+          .from('user_conversations')
+          .select('*')
+          .eq('user_id', userID)
+          .eq('partner_id', donorID)
+          .eq('item_id', itemID)
+          .single();
+
+      if (newUserConversationError) {
+        console.error(
+          'Error fetching new user conversation:',
+          newUserConversationError
+        );
+        return null;
+      }
+      return newUserConversation;
     } catch (error) {
       console.error(error);
+      return null;
     }
   } else {
```

Comment:
following the last comment I think we could remove this `if else` entirely because we would always wrap the call of this function in a `try catch` and if there are no values to pass down to this function we would fail immediately

Reply:
Thanks, I agree 100%, I will try to correct it

Reply:
So I took away the if-else block, since it really is redundant now, and also changed the behavior of the fnc startNewConversation a little 🫣🤪 due to the fact that it is not necessary to return null. The function should return only the conversation object or throw an error if something goes wrong, which we will then process in the try catch block in the clickHandler.